core/lib/spree/core/error_reporter.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Spree
+  module Core
+    module ErrorReporter
+      extend self
+
+      def report(error, severity = :error, metadata = {})
+        reporters.each do |reporter|
+          reporter.report(error, severity, metadata)
+        end
+
+        true
+      end
+
+      ##
+      # All registered reporters
+      # @return [Array<ErrorReporter::Base]
+      #
+      def reporters
+        @reporters ||= []
+      end
+
+      ##
+      # Add a reporter
+      # @param reporter_class [ErrorReporter::Base]
+      # @return [Array<ErrorReporter::Base>]
+      #
+      def add_reporter(reporter_class)
+        unless reporter_class < Spree::Core::ErrorReporter::Base
+          raise ArgumentError, 'Reporter is not type Spree::Core::ErrorReporter::Base'
+        end
```

Comment:
Why require it to be of `ErrorReport::Base`? I would recommend just checking if it `responds_to? :report`. That would be much more in line with Ruby philosophy.